compiler/resolution/functionResolution.cpp

```diff
@@ -7661,7 +7661,7 @@ void printTaskOrForallConstErrorNote(Symbol* aVar) {
     Expr* enclLoop = aVar->defPoint->parentExpr;
 
     USR_PRINT(enclLoop,
-              "The shadow variable '%s' is constant due to forall intents "
+              "The shadow variable '%s' is constant due to intents "
```

Comment:
I wonder if we should consider "loop" intents. Pinging @bradcray to see if he has an opinion about this. I think we may be shifting towards being more open to using just "intent", but wanted to check.

Reply:
Thinking about the general term for the concept, "Loop intents" doesn't feel right since the `begin` and `cobegin` cases aren't loops.  I still find "task intents" to be pretty good except for the concern that `foreach` loops don't technically introduce new tasks (unless we'd want to consider them "hardware tasks"?).  Of course, `for` loops don't introduce tasks either, but once they support intents it still seems like an accurate term in the sense of "the single task executing this loop will get this variable/intent."  "parallelism intents" is another possibility for the general term, though it feels a bit vaguer and like a mouthful.

In any case, I worry that simply saying "due to intents in this loop" feels vague without a modifier (even with the "in this loop" phrase that follows it).  And even the original wording gives me pause today because I wouldn't consider the compiler-introduced shadow variables as being an instance of a task intent (I don't think?).

So, reconsidering the message as a whole, I'd be tempted to say:

> '%s' is constant due to a compiler-introduced shadow variable at this point

for the case that the compiler introduced the shadow variable and 

> '%s' is constant due to the `with`-clause on this statement

when it doesn't.  Can we easily distinguish between those cases (by looking at a FLAG on `aVar`, say, like `FLAG_COMPILER_GENERATED` or `FLAG_TEMP`?).

Reply:
> Can we easily distinguish between those cases (by looking at a FLAG on aVar, say, like FLAG_COMPILER_GENERATED or FLAG_TEMP?)

Running it through the debugger and looking at `viewFlags` on `aVar` I don't see any flag that would let us disambiguate if the variable was coming from a shadow an example where the intent is implicit there's nothing that makes it obvious.

Not to say it can't be done, but I'm inclined just to reword this to say "is constant due to task intents in this loop" and move on (even though as you mentioned 'task' may not be the right word for `foreach` and eventually `for`).

I've spawned a new GH issue for follow up: https://github.com/chapel-lang/chapel/issues/23783

Though unless I get pushback, I'm not really planning to pick this up myself unless I see a way to easily fix it as I continue working on implementing `foreach` intents.  (and presumably this'll whole section of code will be subsumed by Dyno anyway).